.changeset/six-vans-study.md

```diff
@@ -0,0 +1,5 @@
+---
+'@srcbook/api': patch
+---
+
+fix(api): Handle session cleanup after srcbook deletion. Resolved an issue where deleting an srcbook would lead to an uncaught exception due to an orphaned session. The fix ensures proper cleanup of associated sessions when an srcbook is deleted, preventing the "Session with id not found" error.
```

Comment:
Can we simplify this message to something like "Fix uncaught exceptions that can occur after Srcbook deletion" ?

Reply:
This description is better placed in the PR description, with a short message that goes in the Changelog (which will link back to the PR)

Reply:
@benjreinhart sure. But it seems like the issue is much bigger than what I anticipated previously. It looks like a race condition going on between tsserver and actual session deletion which is leading to this error

Reply:
@benjreinhart this simply order change of session deletion won't make it go away. 

The issue, I believe is, not properly handling the errors for onSemanticDiag and onSuggestionDiag event when the session id is not found. 

<img width="563" alt="image" src="https://github.com/user-attachments/assets/f5ff585d-cfd8-4c44-9de8-a81fc9393b14">

At least, the server didn't crash when I added try...catch to catch the errors.